src/profile-logic/marker-data.js

```diff
@@ -116,15 +119,25 @@ export function getSearchFilteredMarkerIndexes(
   if (!searchRegExp) {
     return markerIndexes;
   }
+  const categoryList = getDefaultCategories();
```

Comment:
We need to pass the categories from the profile to the function `getSearchFilteredMarkerIndexes`, indeed they can be different than the ones from `getDefaultCategories`.

This will need to be done in 2 locations:
https://github.com/firefox-devtools/profiler/blob/7d10711d051409f12605020c1bb6b7488bcfbc52/src/selectors/per-thread/markers.js#L288-L295
and
https://github.com/firefox-devtools/profiler/blob/7d10711d051409f12605020c1bb6b7488bcfbc52/src/selectors/per-thread/markers.js#L346-L353

You can get them with this selector (this is an example in the same file):
https://github.com/firefox-devtools/profiler/blob/7d10711d051409f12605020c1bb6b7488bcfbc52/src/selectors/per-thread/markers.js#L419

Reply:
changes made in recent push